### PR TITLE
boards/arm/stm32h7/stm32h747i-disco: bring support for SD card and FAT FS

### PR DESCRIPTION
**Pull request by povauboin**

### CODE REVIEW

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/****************************************************************************
+ * boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c
+ *
+ *   Copyright (C) 2016 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
```
Comment (Ouss4) [inline boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c:8]: New files should have Apache license. Please copy the text from [here](https://github.com/apache/incubator-nuttx/blob/master/sched/sched/sched_getcpu.c)

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c [hunk 2]
```diff
@@ -0,0 +1,175 @@
+/****************************************************************************
+ * boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c
+ *
+ *   Copyright (C) 2016-2019 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
```
Comment (Ouss4) [inline boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c:9]: New files should have Apache license. Please copy the text from [here](https://github.com/apache/incubator-nuttx/blob/master/sched/sched/sched_getcpu.c)

Comment (Ouss4) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c:9]: @patacongo Could you please give your explicit approval for this one.

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c (hunk 1 above)
Comment (patacongo) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c:8]: No 3rd party licensed code is permitted in Apache repositorys.  This cannot be merged without the Apache license header with no author and no copyright.

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c (hunk 2 above)
Comment (patacongo) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c:9]: No, we cannot bring any new 3rd party code into the repository. We can never do that unless they provide a software grant (SGA).

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c (hunk 1 above)
Comment (patacongo) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c:8]: When you contribute software to NuttX you are literally giving it to the Apached Software Foundation and must have the Apache license.

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c (hunk 2 above)
Comment (Ouss4) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c:9]: I meant the approval to change the license to Apache.

Comment (patacongo) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c:9]: Ahh... talking with Abdelatif, I missunderstood the question.  I am the copyright holder and the old SDMMC code and do give my approval to change the license header to the Apache 2.0 license header cited above.

The one with the PX4 copyright must also have an Apache 2.0 header to be considered.

BTW:  The current version of my original SDMMC file does now have an Apache header and also includes requires changes that you do not have in this PR.

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c (hunk 1 above)
Comment (povauboin) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_dma_alloc.c]: license is now changed to Apache

File: boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c (hunk 2 above)
Comment (povauboin) [reply boards/arm/stm32h7/stm32h747i-disco/src/stm32_sdmmc.c]: license is now changed to Apache

**Comment by justinmclean** [comment]
Hi,

How exactly has this license changed? Has the "PX4 Development Team”  given permission and submitted a software grant?

Thanks,
Justin

**Comment by povauboin** [comment]
I copied the file from ./boards/arm/stm32/nucleo-f429zi/src/stm32_dma_alloc.c which has apache license

**Comment by xiaoxiang781216** [comment]
@davids5 should we merge this PR?

**Comment by davids5** [comment]
@xiaoxiang781216  as I understand it there is an unresolved License issue.

**Comment by patacongo** [comment]
Was this file created just for NuttX?  Does it have a history as a PX4 file that is being brought into NuttX?  In this latter case, there this a license issue.  I don't think that the author has the authority to change the license.

On the other hand, If the file was created specifically for NuttX and has never been a part of PX4, then I would say that the creator of the file simply put the wrong license on the file due to a lack of understanding of the license requirements.  In this latter case, it is just a legitimate error and not really a license issue since it was never a part of PX4.  However, if this is the case, then you probably should amend the commit so that it does not appear that it was a PX4 file that has  had its license inappropriately changed.

If merged as is, then this will forever be a licensing problem.

**Comment by davids5** [comment]
Please have a look at @patacongo https://github.com/apache/incubator-nuttx/search?q=stm32_dma_alloc.c&unscoped_q=stm32_dma_alloc.c,

**Comment by xiaoxiang781216** [comment]
> Please have a look at @patacongo https://github.com/apache/incubator-nuttx/search?q=stm32_dma_alloc.c&unscoped_q=stm32_dma_alloc.c,

Can PX4 team give us the permission to chagne the copyright? Otherwise, this PR will block forever.

**Comment by patacongo** [comment]
Can this be merged?  It is passing all of the checks.  I see no references to any unacceptable header files.  Is there any remaining show stopper on this?  If no, I will merge it.

**Comment by patacongo** [comment]
@povauboin It looks to me this PR is ready to be merged.  But now, there is a conflict with the version of arch/arm/src/stm32h7/stm32_sdmmc.c on your branch and the version in the upstream master.

Can you please resolve and and I promise to merge ASAP.

Also.. could you please squash the commits into one or two?

**Comment by patacongo** [comment]
NOTE:  I would have been happy to resolve this conflict for you.  But I do not have access to your branch.  The default action when creating a PR is to allow maintainers to modify your branch.  But since I have no priveleges in your branch, I cannot do that for you.

**Comment by patacongo** [comment]
@povauboin I fixed the conflict.  But since your branch is protected I had create a new branch and rebase with master to fix the conflict.  This PR is now replaced with #1056

**Comment by povauboin** [comment]
@patacongo sorry I missed the emails in the flow. Thanks for resolving the conflict.

**Pull request closed without merging**